### ISSUE TITLE
NoPCM CodeGen "Missing" Files

### ISSUE THREAD

**Issue by Mornix** [JacquesCarette/Drasil#1177]
When a `Choices`' `inputStructure` field is parametrized with `Loose` (specifically this issue is written using NoPCM as the example, however this also occurs if GlassBR is set to `Loose`), the value is passed through the codegen to affect the structure of the generated code. One of the "differences" in `Loose` structure mode is `derived_values()` is generated in `InputParamters.py` rather than the file `DerivedValues.py`.

`InputParameters.py`:
```python
...
T_W = 0.0
E_W = 0.0

def derived_values():
    None

def input_constraints(inParams):
    if (not(inParams.A_C <= A_C_max)) :
        ...
```

In this situation `Control.py` continues to generate the line `DerivedValues.derived_values()`:
```python
...
import DerivedValues

inputfile = sys.argv[1]
inParams = InputParameters.InputParameters()
func_get_input(inputfile, inParams)
DerivedValues.derived_values(inParams)
InputConstraints.input_constraints(inParams)
OutputFormat.write_output(inParams)
```

One possible solution to this issue is the line should read: `InputParameters.derived_values(inParams)` instead.

The issue looks to be with the `eMap` field of `CodeSpec` in which `modExportMap` hardcodes `derived_values` as residing in `DerivedValues` (even though it is not generated!) 

https://github.com/JacquesCarette/Drasil/blob/fff755edfbf71a8650fe6cbbda128ff3a37757d0/code/drasil-code/Language/Drasil/CodeSpec.hs#L231-L240

One implementation-level solution could be to return a modified `CodeSpec` from `genModule` which dynamically adds the correct module definition to `eMap`:
https://github.com/JacquesCarette/Drasil/blob/1432cab83b8011cf4fe7259282324e99969aed4a/code/drasil-code/Language/Drasil/Code/Imperative/Import.hs#L300-L310

**Comment by JacquesCarette** [comment]
I'm sure you're right -- but I do not actually understand this issue. Whoever ought to be assigned to fix this needs to be able to understand the issue.  If you intend to fix it, that's fine. Otherwise, please expand the description so that others can understand it too.

**Comment by Mornix** [comment]
You are correct. This issue seemed to be written as a disjoint set of notes. I've cleaned it up now to hopefully be more cohesive.

**Comment by JacquesCarette** [comment]
It does seem better.

**Comment by bmaclach** [comment]
This issue was fixed with the merge of #1549. Closing.
